packages/hoppscotch-sh-admin/src/components/users/Details.vue

```diff
@@ -0,0 +1,137 @@
+          class="absolute left-16 bottom-0 text-xs font-medium px-3 py-0.5 rounded-full bg-green-900 text-green-300"
+        >
+          {{ t('users.admin') }}
+        </span>
+      </div>
+
+      <div v-if="user.uid">
+        <label class="text-secondaryDark" for="username">{{
+          t('users.uid')
+        }}</label>
+        <div
+          class="w-full p-3 mt-2 bg-divider border-gray-600 rounded-md focus:border-emerald-600 focus:ring focus:ring-opacity-40 focus:ring-emerald-500"
+        >
+          {{ user.uid }}
+        </div>
+      </div>
+      <div>
+        <label class="text-secondaryDark" for="username">{{
+          t('users.name')
+        }}</label>
+        <div
+          class="w-full p-3 mt-2 bg-divider border-gray-600 rounded-md focus:border-emerald-600 focus:ring focus:ring-opacity-40 focus:ring-emerald-500"
+        >
+          <span v-if="user.displayName">
+            {{ user.displayName }}
+          </span>
+          <span v-else> {{ t('users.unnamed') }} </span>
+        </div>
+      </div>
+      <div v-if="user.email">
+        <label class="text-secondaryDark" for="username">{{
+          t('users.email')
+        }}</label>
+        <div
+          class="w-full p-3 mt-2 bg-divider border-gray-600 rounded-md focus:border-emerald-600 focus:ring focus:ring-opacity-40 focus:ring-emerald-500"
+        >
+          {{ user.email }}
+        </div>
+      </div>
+      <div v-if="user.createdOn">
+        <label class="text-secondaryDark" for="username">{{
+          t('users.created_on')
+        }}</label>
+        <div
+          class="w-full p-3 mt-2 bg-divider border-gray-600 rounded-md focus:border-emerald-600 focus:ring focus:ring-opacity-40 focus:ring-emerald-500"
+        >
+          {{ getCreatedDateAndTime(user.createdOn) }}
+        </div>
+      </div>
+    </div>
+
+    <div class="flex justify-start mt-8">
+      <span v-if="!user.isAdmin">
+        <HoppButtonPrimary
+          class="mr-4"
+          filled
+          outline
+          :label="t('users.make_admin')"
+          @click="emit('make-admin', user.uid)"
+        />
+      </span>
+      <span v-else>
+        <HoppButtonPrimary
+          class="mr-4"
+          filled
+          outline
+          :icon="IconUserMinus"
+          :label="t('users.remove_admin_privilege')"
+          @click="emit('remove-admin', user.uid)"
+        />
+      </span>
+      <HoppButtonSecondary
+        v-if="!user.isAdmin"
+        class="mr-4 bg-red-600 text-white hover:text-gray-100"
+        filled
+        outline
+        :label="t('users.delete')"
+        :icon="IconTrash"
+        @click="emit('delete-user', user.uid)"
+      />
+
+      <HoppButtonSecondary
+        v-if="user.isAdmin"
+        class="mr-4 bg-red-600 text-white hover:text-gray-100"
+        filled
+        outline
+        :icon="IconTrash"
+        :label="t('users.delete')"
+        @click="toast.error(t('state.remove_admin_to_delete_user'))"
+      />
```

Comment:
These can be unified as well. The props can be dynamically bound based on the `user.isAdmin` check.